python/cudf/cudf/core/df_protocol.py

```diff
@@ -645,7 +644,13 @@ def __dataframe__(
 _INTS = {8: cp.int8, 16: cp.int16, 32: cp.int32, 64: cp.int64}
 _UINTS = {8: cp.uint8, 16: cp.uint16, 32: cp.uint32, 64: cp.uint64}
 _FLOATS = {32: cp.float32, 64: cp.float64}
-_CP_DTYPES = {0: _INTS, 1: _UINTS, 2: _FLOATS, 20: {8: bool}}
+_CP_DTYPES = {
+    0: _INTS,
+    1: _UINTS,
+    2: _FLOATS,
+    20: {8: bool},
+    21: {8: cp.uint8},
```

Comment:
Strings columns in the protocol are composed of a `uint8` buffer of `data` (chars) and an `int` buffer of `offsets`. The `data` column has a distinct type ID (`21`) that we previously didn't handle.

As a clarifying example, look at the buffers returned by a Pandas dataframe containing a single string column:

```python
In [17]: pd.DataFrame({'a': ['x', 'y', 'z']}).__dataframe__().get_column(0).get_buffers()
Out[17]: 
{'data': (PandasBuffer({'bufsize': 3, 'ptr': 140001666253824, 'device': 'CPU'}),
  (<DtypeKind.STRING: 21>, 8, 'u', '=')),
 'validity': (PandasBuffer({'bufsize': 3, 'ptr': 94467671186912, 'device': 'CPU'}),
  (<DtypeKind.BOOL: 20>, 8, 'b', '=')),
 'offsets': (PandasBuffer({'bufsize': 32, 'ptr': 94467669930400, 'device': 'CPU'}),
  (<DtypeKind.INT: 0>, 64, 'l', '='))}
```